src/OpenTelemetry.Extensions.PersistentStorage/PersistentStorageEventSource.cs

```diff
@@ -16,92 +16,54 @@
 
 using System;
 using System.Diagnostics.Tracing;
-using System.Runtime.CompilerServices;
+using System.Globalization;
+using System.Threading;
 
 namespace OpenTelemetry.Extensions.PersistentStorage
 {
     [EventSource(Name = EventSourceName)]
     internal sealed class PersistentStorageEventSource : EventSource
     {
         public static PersistentStorageEventSource Log = new PersistentStorageEventSource();
-        private const string EventSourceName = "OpenTelemetry-Contrib-PersistentStorage";
+        private const string EventSourceName = "OpenTelemetry-Extensions-PersistentStorage";
 
         [NonEvent]
-        public void Critical(string message, object value = null)
+        public void PersistentStorageException(string className, string message, Exception ex)
         {
-            this.Write(EventLevel.Critical, message, value);
-        }
-
-        [NonEvent]
-        public void Error(string message, object value = null)
-        {
-            this.Write(EventLevel.Error, message, value);
-        }
-
-        [NonEvent]
-        public void Warning(string message, object value = null)
-        {
-            this.Write(EventLevel.Warning, message, value);
+            if (this.IsEnabled(EventLevel.Error, EventKeywords.All))
+            {
+                this.PersistentStorageException(className, message, ToInvariantString(ex));
+            }
         }
 
-        [NonEvent]
-        public void Informational(string message, object value = null)
+        [Event(1, Message = "{0}: {1}: {2}", Level = EventLevel.Error)]
+        public void PersistentStorageException(string className, string message, string ex)
```

Comment:
What is the semantic of `PersistentStorageException`? (e.g. who is the faulty party, who should take action)

Reply:
E.g. who will be consuming these logs, what should they expect to do with these logs, are these logs helpful to drive any meaningful action?